pyomo/contrib/interior_point/interface.py

```diff
@@ -0,0 +1,624 @@
+        return self._nlp.n_primals()
+
+    def nnz_hessian_lag(self):
+        return self._nlp.nnz_hessian_lag()
+
+    def set_obj_factor(self, obj_factor):
+        self._nlp.set_obj_factor(obj_factor)
+
+    def get_obj_factor(self):
+        return self._nlp.get_obj_factor()
+
+    def n_eq_constraints(self):
+        return self._nlp.n_eq_constraints()
+
+    def n_ineq_constraints(self):
+        return self._nlp.n_ineq_constraints()
+
+    def nnz_jacobian_eq(self):
+        return self._nlp.nnz_jacobian_eq()
+
+    def nnz_jacobian_ineq(self):
+        return self._nlp.nnz_jacobian_ineq()
+
+    def init_primals(self):
+        primals = self._nlp.init_primals()
+        return primals
+
+    def init_slacks(self):
+        slacks = self._nlp.evaluate_ineq_constraints()
+        return slacks
+
+    def init_duals_eq(self):
+        return self._nlp.init_duals_eq()
+
+    def init_duals_ineq(self):
+        return self._nlp.init_duals_ineq()
+
+    def init_duals_primals_lb(self):
+        return self._init_duals_primals_lb
+
+    def init_duals_primals_ub(self):
+        return self._init_duals_primals_ub
+
+    def init_duals_slacks_lb(self):
+        return self._init_duals_slacks_lb
+
+    def init_duals_slacks_ub(self):
+        return self._init_duals_slacks_ub
+
+    def set_primals(self, primals):
+        self._nlp.set_primals(primals)
+
+    def set_slacks(self, slacks):
+        self._slacks = slacks
+
+    def set_duals_eq(self, duals):
+        self._nlp.set_duals_eq(duals)
+
+    def set_duals_ineq(self, duals):
+        self._nlp.set_duals_ineq(duals)
+
+    def set_duals_primals_lb(self, duals):
+        self._duals_primals_lb = duals
+
+    def set_duals_primals_ub(self, duals):
+        self._duals_primals_ub = duals
+
+    def set_duals_slacks_lb(self, duals):
+        self._duals_slacks_lb = duals
+
+    def set_duals_slacks_ub(self, duals):
+        self._duals_slacks_ub = duals
+
+    def get_primals(self):
+        return self._nlp.get_primals()
+
+    def get_slacks(self):
+        return self._slacks
+
+    def get_duals_eq(self):
+        return self._nlp.get_duals_eq()
+
+    def get_duals_ineq(self):
+        return self._nlp.get_duals_ineq()
+
+    def get_duals_primals_lb(self):
+        return self._duals_primals_lb
+
+    def get_duals_primals_ub(self):
+        return self._duals_primals_ub
+
+    def get_duals_slacks_lb(self):
+        return self._duals_slacks_lb
+
+    def get_duals_slacks_ub(self):
+        return self._duals_slacks_ub
+
+    def primals_lb(self):
+        return self._nlp.primals_lb()
+
+    def primals_ub(self):
+        return self._nlp.primals_ub()
+
+    def ineq_lb(self):
+        return self._nlp.ineq_lb()
+
+    def ineq_ub(self):
+        return self._nlp.ineq_ub()
+
+    def set_barrier_parameter(self, barrier):
+        self._barrier = barrier
+
+    def pyomo_nlp(self):
+        return self._nlp
+
+    def evaluate_primal_dual_kkt_matrix(self, timer=None):
```

Comment:
Can this be used to extract the KKT system of the barrier problem at x*, lambda*? If so this matrix and the rhs is all that is needed to compute the inverse of the reduced hessian in a similar fashion to what it is in #1282

Reply:
Yes, it can. Good point.